### ISSUE TITLE
Boolean Trip logging speed filter

### ISSUE THREAD

**Issue by sonora** [osmandapp/Osmand#3238]
The GPS output usually reports a speed value for when a device is detected as in motion, and omits this if not. This can be used as an intrinsic motion detection in our trip recording.

In our trip logging filters, we do not currently support to tap into this feature. I suggest we rename our current "Logging minimum speed" setting to

`"Logging minimum speed - Log only points where a speed value is reported."`

and make it a boolean setting. Filtering if a speed>0 is reported for a point is much ore useful than a float setting with 1/2/3/4/5/6/7 kmh cutoffs, where it is unclear to the user what to select at all.

**Comment by vshcherb** [comment]
I took a look at other track recorders and this is what I've found so far. So I can't agree or disagree with you, though I have 2 main points:
1) Users asked for that setting
2) Similar apps have that interpretation

**Comment by sonora** [comment]
Yes, some software out there with many different approaches - not all of them make sense. :)

AllI'm saying is: From what I see the desired functionality in the vast majority of cases is simply "motion detection", nothing more sophisticated. In OsmAnd we have now one good way to achieve that, it is the displacement filter checking if your position changed by more than x meters.

The second easy way to detect motion is if the GPS chipset reports a speed >0, and this we do not currently facilitate to filter.

**Comment by vshcherb** [comment]
I think @xmd5a2  would disagree with you and me too, on the bicycle there is a reasonable speed of 5 kmh, everything less is just a noise of you are coming to the traffic signals or trying to cross the river. May be I'm wrong.

**Comment by xmd5a2** [comment]
Yes, it can be very useful to filter out very slow motion when riding bicycle. As Victor said bicyclist can cross a river. This is usually long process and GPS points recorded during it are usually not needed.

**Comment by sonora** [comment]
I guess we must be talking strictly the spatial dimension then, not things like the time domain or speed domain (which enters also average speed analysis etc.)

Hello Leonid, out of scientific curiosity I would be interested in seeing such a track where you think noise is present predominantly where 0 <speed <5km/h, and with which device it was recorded. Would you have one you could maybe mail me? Thx- Hardy

**Comment by sonora** [comment]
@vshcherb , @xmd5a2 : It is a little hard to give a lecture on signal processing in the framework of an issue.

For future reference, and maybe some lessons learned for our code and presets, I have prepared a little white paper here: http://osmand.net/features?id=trip-recording-plugin#Trip_Logging:_Some_Basics_on_Parameters_and_Filtering. Maybe we want to come back to this later and improve our settings or presets with information and some conclusions I have documented there.  :)
